charts/monochart/templates/daemonset.yaml

```diff
@@ -54,14 +54,18 @@ spec:
         securityContext:
 {{ toYaml . | indent 10 }}
 {{- end }}
-{{- $defaultService := (get .Values.service "default") }}
-{{ if $defaultService }}
         ports:
-{{- range $name, $port := (get $defaultService "ports") }}
-{{- if $port }}
+{{- $registeredPorts := list -}}
+{{- range $serviceName, $service := .Values.service }}
+{{- $portId := (sha256sum (toString $service)) -}}
```

Comment:
Use `sha256sum` to generate a footprint of the port spec. This way we don't add ports to the list if they are identical